source/custom/nimxlcterminaladaptor_restricted_converters.h

```diff
@@ -31,20 +29,39 @@ struct NIErrStatusOutputConverter {
     return &status;
   }
 
+  std::string UrlEncode(char* input) const
```

Comment:
nit: there are all kinds of different standards once you include imports and generated code. But in hand-written c++ in `grpc-device` method names are `snake_case`.